src/main/java/org/broadinstitute/hellbender/tools/spark/sv/evidence/EvidenceTargetLinkClusterer.java

```diff
@@ -0,0 +1,122 @@
+                    final SVIntervalTree.Entry<EvidenceTargetLink> sourceIntervalEntry = it.next();
+                    final EvidenceTargetLink oldLink = sourceIntervalEntry.getValue();
+                    // todo: what to do if there are more than one distal targets -- for now just taking the first one
+                    if (nextEvidence.hasDistalTargets(readMetadata) &&
+                            strandsMatch(nextEvidence.isForwardStrand(), sourceIntervalEntry.getValue().sourceForwardStrand)
+                        && (nextEvidence.getDistalTargets(readMetadata).get(0).overlaps(oldLink.target) &&
+                                strandsMatch(nextEvidence.getDistalTargetStrands(readMetadata).get(0), oldLink.targetForwardStrand))) {
+                            // if it does, intersect the source and target intervals to refine the link
+                            it.remove();
+                            final SVInterval newSource = sourceIntervalEntry.getInterval().intersect(nextEvidence.getLocation());
+                            final SVInterval newTarget = oldLink.target.intersect(nextEvidence.getDistalTargets(readMetadata).get(0));
+                            updatedLink = new EvidenceTargetLink(newSource,
+                                    oldLink.sourceForwardStrand,
+                                    newTarget,
+                                    oldLink.targetForwardStrand,
+                                    nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                            ? oldLink.splitReads : oldLink.splitReads + 1,
+                                    nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                            ? oldLink.readPairs + 1 : oldLink.readPairs);
+                            break;
+                    }
+                }
+                if (updatedLink == null) {
+                    updatedLink = new EvidenceTargetLink(
+                            nextEvidence.getLocation(),
+                            nextEvidence.isForwardStrand(),
+                            nextEvidence.getDistalTargets(readMetadata).get(0),
+                            nextEvidence.getDistalTargetStrands(readMetadata).get(0),
+                            nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                    ? 0 : 1,
+                            nextEvidence instanceof BreakpointEvidence.DiscordantReadPairEvidence
+                                    ? 1 : 0);
+                }
+                currentIntervalsWithTargets.put(updatedLink.source, updatedLink);
+            }
+        }
+
+        for (Iterator<SVIntervalTree.Entry<EvidenceTargetLink>> it = currentIntervalsWithTargets.iterator(); it.hasNext(); ) {
```

Comment:
SVIntervalTree implements Iterable, so this could be expressed a little more tersely if you wanted to:
```
        currentIntervalsWithTargets.forEach(entry -> links.add(entry.getValue()));
``

Reply:
done